pennylane/noise/conditionals.py

```diff
@@ -0,0 +1,481 @@
+from pennylane.templates import ControlledSequence
+from pennylane.wires import WireError, Wires
+
+# pylint: disable = unnecessary-lambda, too-few-public-methods
+
+
+class WiresIn(BooleanFn):
+    """A conditional for evaluating if a given wire exist in a specified set of wires.
+
+    Args:
+        wires (Union[Iterable[int, str], Wires]): sequence of wires for building the wire set.
+
+    .. seealso:: Users are advised to use :func:`~.wires_in` for a functional construction.
+    """
+
+    def __init__(self, wires):
+        self._cond = set(wires)
+        self.condition = self._cond
+        super().__init__(
+            lambda wire: _get_wires(wire).issubset(self._cond), f"WiresIn({list(wires)})"
+        )
+
+
+class WiresEq(BooleanFn):
+    """A conditional for evaluating if a given wire is equal to a specified set of wires.
+
+    Args:
+        wires (Union[Iterable[int, str], Wires]): sequence of wires for building the wire set.
+
+    .. seealso:: Users are advised to use :func:`~.wires_eq` for a functional construction.
+    """
+
+    def __init__(self, wires):
+        self._cond = set(wires)
+        self.condition = self._cond
+        super().__init__(
+            lambda wire: _get_wires(wire) == self._cond,
+            f"WiresEq({list(wires) if len(wires) > 1 else list(wires)[0]})",
+        )
+
+
+def _get_wires(val):
+    """Extract wires as a set from an integer, string, Iterable, Wires or Operation instance.
+
+    Args:
+        val (Union[int, str, Iterable, ~.wires.Wires, ~.operation.Operation]): object to be used
+            for building the wire set.
+
+    Returns:
+        set[Union[int, str]]: computed wire set
+
+    Raises:
+        ValueError: if the wire set cannot be computed for ``val``.
+    """
+    iters = val if isinstance(val, (list, tuple, set, Wires)) else getattr(val, "wires", [val])
+    try:
+        wires = set().union(*((getattr(w, "wires", None) or Wires(w)).tolist() for w in iters))
+    except (TypeError, WireError):
+        raise ValueError(f"Wires cannot be computed for {val}") from None
+    return wires
+
+
+def wires_in(wires):
+    """Builds a conditional as a boolean function for evaluating
+    if a given wire exist in a specified set of wires.
+
+    Args:
+        wires (Union(Iterable[int, str], Wires, Operation)): object to be used
+            for building the wire set.
+
+    Returns:
+        :class:`WiresIn <pennylane.noise.conditionals.WiresIn>`: a callable object that
+        accepts an argument ``wire`` and gives a boolean output. It evaluates to ``True``
+        if the wire set constructed from ``wire`` is a subset of the one built from the
+        specified set.
+
+    Raises:
+        ValueError: if the wire set cannot be computed from ``wires``.
+
+    **Example**
+
+    One may use ``wires_in`` with a given sequence of wires which are used as a wire set:
+
+    >>> cond_func = qml.noise.wires_in([0, 1])
+    >>> cond_func(0)
```

Comment:
```suggestion
    >>> cond_func(qml.X(0))
```
Preference to show the operator-based input as that's more standard for what we'd expect the boolean functions to have input as part of the noise model